ObjectiveGit/GTIndex.m

```diff
@@ -336,10 +344,58 @@ - (BOOL)enumerateConflictedFilesWithError:(NSError **)error usingBlock:(void (^)
 	BOOL shouldAbortImmediately;
 };
 
+- (BOOL)addPathspecs:(NSArray *)pathspecs flags:(GTIndexAddOptionFlags)flags error:(NSError **)error passingTest:(GTIndexPathspecMatchedBlock)block {
+	NSAssert(self.repository.isBare == NO, @"This method only works with non-bare repositories.");
+
+	__block git_strarray strarray = pathspecs.git_strarray;
+	@onExit {
+		if (strarray.count > 0) git_strarray_free(&strarray);
+	};
+
+	struct GTIndexPathspecMatchedInfo payload = {
+		.block = block,
+		.shouldAbortImmediately = NO,
+	};
+
+	int returnCode = git_index_add_all(self.git_index, &strarray, (unsigned int)flags, (block != nil ? GTIndexPathspecMatchFound : NULL), &payload);
```

Comment:
I realize it's not from here, but `GTIndexPathspecMatchFound` sounds strange (I was wondering if it was a snuck-in flag or something).

Can you add `Callback` to the function name ?